backend.go

```diff
@@ -128,6 +133,54 @@ func (r *RedisBackend) IncrementID(ctx context.Context, id int) (int64, error) {
 	return cnt, nil
 }
 
+func (r *RedisBackend) GetMessageReply(ctx context.Context, msg MessageIdentifier) ([]Message, error) {
+	log.Info().Str("return_queue", msg.Retqueue).Msg("Waiting reply")
+	responses := []Message{}
+	retQueueLen, err := r.client.LLen(ctx, msg.Retqueue).Result()
+	if err != nil {
+		log.Error().Err(err).Msg("error fetching from redis")
+		return responses, err
+	}
+	// TODO: check if the message ready
+	if int(retQueueLen) < 1 {
+		return nil, errors.New("The message not handled yet, it is in progress.")
+	} else {
+		// loop and return the list
+		results, err := r.client.LRange(ctx, msg.Retqueue, 0, -1).Result()
+		if err != nil {
+			log.Error().Err(err).Msg("error fetching from redis")
+			return responses, err
+		}
+		for _, msgJSON := range results {
+			responseMsg := Message{}
+			if err := json.Unmarshal([]byte(msgJSON), &responseMsg); err != nil {
+				log.Error().Err(err).Msg("error decoding entry from redis")
+				break
+			}
+			decoded, err := base64.StdEncoding.DecodeString(responseMsg.Data)
+			if err != nil {
+				log.Error().Err(err).Msg("failed to decode response message data")
```

Comment:
break or continue if an error happens. And I think both error handlings should be `continue`, because if we introduced a breaking change to the message types, the old ones should be ignored and the proper ones returned.